### PR TITLE
[dif/otp_ctrl] Add sanity tests for otp_ctrl DIFs.

### PR DESCRIPTION
**Pull request by mcy**
~~Currently blocked by #4074 (pretty cool that this revealed a HW bug, though).~~

### CODE REVIEW

File: sw/device/tests/dif/dif_otp_ctrl_sanitytest.c [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+static const char kTestData[] = "abcdabcdabcdabc";
+_Static_assert(ARRAYSIZE(kTestData) % sizeof(uint32_t) == 0,
+               "kTestData must be a word array");
+
+const test_config_t kTestConfig;
+
+/**
+ * Busy-wait until the DAI is done with whatever operation it is doing.
+ */
+void wait_for_dai() {
```
Comment (silvestrst) [inline sw/device/tests/dif/dif_otp_ctrl_sanitytest.c:27]: `void wait_for_dai()` -> `static void wait_for_dai()`?

File: sw/device/tests/dif/dif_otp_ctrl_sanitytest.c [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    LOG_INFO("Waiting for DAI...");
+    dif_otp_ctrl_status_t status;
+    CHECK(dif_otp_ctrl_get_status(&otp, &status) == kDifOtpCtrlOk);
+    if (bitfield_bit32_read(status.codes, kDifOtpCtrlStatusCodeDaiIdle)) {
+      return;
+    }
+  }
+}
+
+bool test_main(void) {
```
Comment (silvestrst) [inline sw/device/tests/dif/dif_otp_ctrl_sanitytest.c:38]: Could you please add description somewhere of what this code is testing for?

**Comment by mcy** [comment]
Should now work on the "real" hardware! =)

**Review by msfschaffner — APPROVED** [review]
Cool :), LGTM!

**Comment by imphil** [comment]
A good example of GCC finding something clang didn't :smile: 

```
FAILED: sw/device/tests/dif/64cbfc5@@dif_otp_ctrl_sanitytest_lib@sta/dif_otp_ctrl_sanitytest.c.o 
/opt/buildcache/riscv/bin/riscv32-unknown-elf-gcc -Isw/device/tests/dif/64cbfc5@@dif_otp_ctrl_sanitytest_lib@sta   -I/home/vsts/work/1/s -I/home/vsts/work/1/a/build-out -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=c11 -g -nostdinc -isystem/home/vsts/work/1/s/sw/device/lib/base/freestanding -fno-asynchronous-unwind-tables -fno-common -Wimplicit-fallthrough -Wswitch-default -Wno-error=unused-function -Wno-unused-parameter -Wno-sign-compare -Wno-missing-field-initializers -Os -fvisibility=hidden -Wa,--no-pad-sections -Wstrict-prototypes -march=rv32imc -mabi=ilp32 -mcmodel=medany -MD -MQ 'sw/device/tests/dif/64cbfc5@@dif_otp_ctrl_sanitytest_lib@sta/dif_otp_ctrl_sanitytest.c.o' -MF 'sw/device/tests/dif/64cbfc5@@dif_otp_ctrl_sanitytest_lib@sta/dif_otp_ctrl_sanitytest.c.o.d' -o 'sw/device/tests/dif/64cbfc5@@dif_otp_ctrl_sanitytest_lib@sta/dif_otp_ctrl_sanitytest.c.o' -c ../../s/sw/device/tests/dif/dif_otp_ctrl_sanitytest.c
In file included from ../../s/sw/device/tests/dif/dif_otp_ctrl_sanitytest.c:12:
../../s/sw/device/tests/dif/dif_otp_ctrl_sanitytest.c: In function 'test_main':
../../s/sw/device/tests/dif/dif_otp_ctrl_sanitytest.c:62:15: error: comparison between 'dif_otp_ctrl_dai_result_t' {aka 'enum dif_otp_ctrl_dai_result'} and 'enum dif_otp_ctrl_result' [-Werror=enum-compare]
   62 |     CHECK(err == kDifOtpCtrlOk, "Failed to program word kTestData[%d] = 0x%8x.",
      |               ^~
/home/vsts/work/1/s/sw/device/lib/testing/check.h:29:11: note: in definition of macro 'CHECK'
   29 |     if (!(condition)) {                                   \
      |           ^~~~~~~~~
```

Maybe there's a clang flag to be added?

**Comment by mcy** [comment]
Gah! I thought I fixed that. At least my test passed in CI.

**Review by gkelly — APPROVED** [review]
Can you please rename to `_smoketest.c`: https://github.com/lowRISC/opentitan/pull/4191

**Pull request merged**